syntax  = "proto3";

package arrus.proto;

import "io/proto/common/IntervalDouble.proto";
import "io/proto/common/IntervalInteger.proto";


message ProbeModel {
  message Id {
    string manufacturer = 1;
    string name = 2;
  }
  Id id = 1;
  repeated uint32 n_elements = 2;
  repeated double pitch = 3;
  IntervalDouble tx_frequency_range = 4;
  // Acceptable voltage range +/-, [0.5*Vpp]
  IntervalInteger voltage_range = 5;
  // Curvature radius; 0 means no curvature.
  double curvature_radius = 6;
}